modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtGetFuture.java

```diff
@@ -340,7 +342,14 @@ private boolean map(KeyCacheObject key, boolean forceKeys) {
             if (parts == null || !F.contains(parts, part.id())) {
                 // By reserving, we make sure that partition won't be unloaded while processed.
                 if (part.reserve()) {
-                    if (forceKeys || (part.state() == OWNING || part.state() == LOST)) {
+                    boolean mvccNoOwners = false;
+
+                    // Force key request is disabled for MVCC. So if there are no partition owners for the given key
+                    // we need to set flag forceKeys to true to avoid useless remapping to other non-owning partitions.
+                    if (cctx.mvccEnabled())
+                        mvccNoOwners = cctx.topology().owners(cctx.affinity().partition(key), topVer).isEmpty();
+
+                    if (forceKeys || (part.state() == OWNING || part.state() == LOST) || mvccNoOwners) {
```

Comment:
Should partition state be LOST if it has no owners?
Is additional flag really needed here?